src/aspire/nufft/finufft.py

```diff
@@ -1,94 +1,87 @@
-        #         (Potentially anything that used this, for example).
-        #       I would error, but I know a bunch of code ASPIRE wants to work,
-        #         the cov2d tutorial for example, would fail and require hacks to run
-        #         if I was strict about dtypes rn.
-        #       I would rather deal with that in other, more targeted PRs.
-        #       This preserves the legacy behavior of admitting singles,
-        #         but I will correct it slightly, to return the precision given as input.
-        #       This approach should contain the hacks here to a single place on the edge,
-        #         instead of spread through the code. ASPIRE code should focus on being
-        #         internally consistent.
-        #       Admittedly not ideal, but ignoring these problems wasn't sustainable.
-
-        self.cast_output = False
-        if self.dtype != np.float64:
-            logger.debug(
-                "This version of finufftpy is hardcoded to doubles internally"
-                "  casting input to doubles, results cast back to singles."
-            )
-            self.cast_output = True
-            self.dtype = np.float64
+        self.dtype_str = dtype_legacy_string(self.dtype)
 
         self.complex_dtype = complex_type(self.dtype)
 
-        # TODO: Things get messed up unless we ensure a 'C' ordering here - investigate why
-        self.fourier_pts = np.asarray(
-            np.mod(fourier_pts + np.pi, 2 * np.pi) - np.pi, order="C", dtype=self.dtype
+        self.fourier_pts = np.ascontiguousarray(
+            np.mod(fourier_pts + np.pi, 2 * np.pi) - np.pi
         )
+
         self.num_pts = fourier_pts.shape[1]
-        self.epsilon = epsilon
 
-        # Get a handle on the appropriate 1d/2d/3d forward transform function in finufftpy
-        self.transform_function = getattr(finufftpy, f"nufft{self.dim}d2{manystr}")
+        self.epsilon = max(epsilon, np.finfo(self.dtype).eps)
+        if self.epsilon != epsilon:
+            logger.debug(
+                f"FinufftPlan adjusted eps={self.epsilon}" f" from requested {epsilon}."
+            )
+
+        self._transform_plan = finufft.Plan(
+            nufft_type=2,
+            n_modes_or_dim=self.sz,
+            eps=self.epsilon,
+            n_trans=self.ntransforms,
+            dtype=self.dtype_str,
+        )
+
+        self._adjoint_plan = finufft.Plan(
+            nufft_type=1,
+            n_modes_or_dim=self.sz,
+            eps=self.epsilon,
+            n_trans=self.ntransforms,
+            dtype=self.dtype_str,
+        )
 
-        # Get a handle on the appropriate 1d/2d/3d adjoint function in finufftpy
-        self.adjoint_function = getattr(finufftpy, f"nufft{self.dim}d1{manystr}")
+        # FINUFFT expects column major signal data,
+        #   but we don't want to transpose large arrays.
+        #   We instead reverse the references to the points axes.
+        self._transform_plan.setpts(*self.fourier_pts[::-1])
```

Comment:
Isn't this already handled by the Finufft Python interface?

Reply:
Yes it is. I replaced the ordering of frequencies in any code under test, and removed this here and in the cufinufft class.